test/integration/job/job_test.go

```diff
@@ -3191,6 +3192,86 @@ func TestFinalizersClearedWhenBackoffLimitExceeded(t *testing.T) {
 	validateNoOrphanPodsWithFinalizers(ctx, t, clientSet, jobObj)
 }
 
+func TestFinalizersNotClearedForPodsNotOwnedJob(t *testing.T) {
```

Comment:
TBH, I'm not sure this is the best way to test it, but it was relatively easy, and it is agnostic of the layer at which the fix exists. However, maybe a unit test for `syncOrphanPod` would be enough. I'm happy to consider alternatives

Reply:
I tentatively remove the integration test in the last commit https://github.com/kubernetes/kubernetes/pull/124798/commits/77b26b1736e1365b22048502fee4e83b4e37625d. I'm ok to put it back if there are voices from reviewers to do so, but for now it seems excessive to me.

Reply:
yeah, I also think a unit should be enough

Reply:
In that case I squash the comments for a cleaner history.